pkg/gitprovider/gitlab.go

```diff
@@ -576,11 +576,12 @@ func (g *GitLabGitProvider) ParseEventData(request *http.Request) (*GitEventData
 	return gitEventData, nil
 }
 
-func (g *GitLabGitProvider) FormatError(response *gitlab.Response) error {
-	body, err := io.ReadAll(response.Body)
-	if err != nil {
-		return fmt.Errorf("status code: %d err: failed to read response body: %v", response.StatusCode, err)
+func (g *GitLabGitProvider) FormatError(err error) error {
+	re := regexp.MustCompile(`([A-Z]+)\s(https:\/\/\S+):\s(\d{3})\s(\{message:\s\d{3}\s.+\})`)
+	match := re.FindStringSubmatch(err.Error())
+	if len(match) > 2 {
```

Comment:
Why do we check only for 2 elements but access 4 of them?

I think the same thing applies to other Format methods.